src/write_concern.ts

```diff
@@ -88,7 +88,17 @@ export class WriteConcern {
     const parentOpts: WriteConcern | WriteConcernSettings | undefined =
       inherit instanceof WriteConcern ? inherit : inherit.writeConcern;
 
-    const { w, wtimeout, j, fsync, journal, wtimeoutMS } = { ...parentOpts, ...opts };
+    const {
+      w = undefined,
```

Comment:
why is the explicit defaulting necessary?

Reply:
Turns out it isn't. I was getting "error TS2525: Initializer provides no value for this binding element and the binding element has no default value" because the opts variable was of type `any` so it TS now warns that there isn't information about those properties from the variables being spread. The better solution is to correctly annotate `opts`.